mycpp/marksweep_heap.cc

```diff
@@ -0,0 +1,114 @@
+#include "mycpp/runtime.h"
+
+void MarkSweepHeap::Init(int max_heap_bytes) {
```

Comment:
It seems like there are a few interface mis-matches, like this one, and the `Collect(num_bytes)`, etc.

Maybe put a TODO there

Reply:
I'm not sure what you mean by 'interface mismatches' here.  Is it that the variable names are different?

The `Collect()` interface is semantically different, but there's no way to make them identical.  I put an assertion and a big block comment to explain it.